carbon-design-system/carbon
# chore(test): add baseline e2e tests and percy snapshots

Refs #8911

Adds baseline end to end tests for 5 components: `AspectRatio`, `Breadcrumb`, `Button`, `ButtonSet`, `Modal`. These baseline tests include a percy snapshot for each component. 

`Modal` got some special treatment as an example for the future of how to test and snapshot components at the different breakpoint widths. Even with this, the modal tests all together only take .75s to execute!

#### Changelog

**New**

- test: e2e and percy snapshots

#### Testing / Reviewing

- Status checks should pass
- Percy status check (and UI) should register new snapshots for the 5 components listed in this PR.

If you're interested to see what the tests are doing: 
  - pull down this branch
  - cd into `packages/react`
  - run `yarn cypress open-ct`
  - this will open the cypress gui in a new instance of chrome where you can select and run tests, and view the UI it's snapshotting and testing against.
